crates/sargon/src/signing/signatures_outecome_types/signatures_outcome.rs

```diff
@@ -37,13 +37,14 @@ impl SignaturesOutcome {
             .into_iter()
             .collect::<IndexSet<_>>();
 
-        let successful_hashes: IndexSet<IntentHash> = successful_transactions
-            .transactions
-            .keys()
-            .cloned()
-            .collect();
-
-        let failure_hashes: IndexSet<IntentHash> =
+        let successful_hashes: IndexSet<TransactionIntentHash> =
+            successful_transactions
+                .transactions
+                .keys()
+                .cloned()
+                .collect();
```

Comment:
lets remove the `:IndexSet<TransactionIntentHash>` and use `collect::<IndexSet<_>>();`  instead